src/Service/UserTransformer.php

```diff
@@ -90,4 +104,33 @@ public function transform(
         }
         return $entity;
     }
-}
+
+    /**
+     * @param UserDTO $dto
+     * @return User
+     * @throws NotValidOldPassword
+     * @throws EntityNotFound
+     */
+    public function changePasswordTransform(
+        UserDTO $dto
+    ): User {
+        $entity = $this->userRepo->find($dto->id);
```

Comment:
On change password you should have only the passwords in the requests. The user id is sent in URL. Pass the $user object when calling `changePasswordTransform()`